backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -212,15 +212,47 @@ public LiveOktaRepository(
           "Cannot update Okta user with unrecognized username");
     }
     User user = users.single();
+    updateUser(user, userIdentity);
+
+    return getOrganizationRoleClaimsForUser(user);
+  }
+
+  private void updateUser(User user, IdentityAttributes userIdentity) {
     user.getProfile().setFirstName(userIdentity.getFirstName());
     user.getProfile().setMiddleName(userIdentity.getMiddleName());
     user.getProfile().setLastName(userIdentity.getLastName());
     // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they
     // don't have regular suffix? You decide.
     user.getProfile().setHonorificSuffix(userIdentity.getSuffix());
     user.update();
+  }
 
-    return getOrganizationRoleClaimsForUser(user);
+  public void reprovisionUser(IdentityAttributes userIdentity) {
+    UserList users = _client.listUsers(userIdentity.getUsername(), null, null, null, null);
+    if (users.stream().count() == 0) {
+      throw new IllegalGraphqlArgumentException(
+          "Cannot reprovision Okta user with unrecognized username");
+    }
+    User user = users.single();
+    UserStatus userStatus = user.getStatus();
+
+    // any org user "deleted" through our api will be in SUSPENDED state
+    if (userStatus != UserStatus.SUSPENDED) {
+      throw new IllegalGraphqlArgumentException(
+          "Cannot reprovision user in unsupported state: " + userStatus);
+    }
+
+    updateUser(user, userIdentity);
+
+    // reset all MFA factors (this triggers an email to the user, which can be configured in Okta)
+    user.resetFactors();
+
+    // transitioning from SUSPENDED -> DEPROVISIONED -> ACTIVE will reset the user's password
+    // and password reset question.  `.activate(true)` will send an activation email (just like
+    // creating a new account).  This cannot be done with `.reactivate()` because `.reactivate()`
+    // requires  the user to be in PROVISIONED state
+    user.deactivate();
+    user.activate(true);
```

Comment:
this works? the okta docs say that `activate` can only be called on users in `STAGED` state.

Reply:
It's actually `STAGED` or `DEPROVISIONED` and `deactivate()` puts the user into `DEPROVISIONED` state.  Maybe this was changed recently?

https://developer.okta.com/docs/reference/api/users/#activate-user